File: plop-templates/adapter/acquiring/src/config/properties/RestTemplateProperties.java
package com.rbkmoney.adapterxpatemerit.config.properties;

import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.context.annotation.Configuration;
import org.springframework.validation.annotation.Validated;

import javax.validation.constraints.NotNull;

@Getter
@Setter
@Validated
@Configuration
@ConfigurationProperties("rest-template")
public class RestTemplateProperties {

    @NotNull
    private int maxTotalPooling;

    @NotNull
    private int defaultMaxPerRoute;

    @NotNull
    private int requestTimeout;

    @NotNull
    private int poolTimeout;

    @NotNull
    private int connectionTimeout;
}

File: plop-templates/adapter/payout/src/config/properties/RestTemplateProperties.kt
package com.rbkmoney.{{adapterPayoutPackageCase bank_name}}.config.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.boot.context.properties.ConstructorBinding
import org.springframework.validation.annotation.Validated
import javax.validation.constraints.NotEmpty

{{#if_eq doc true}}
/**
 * Проперти для restTemplate.
 */
{{/if_eq}}
@Validated
@ConstructorBinding
@ConfigurationProperties(prefix = "rest-template")
data class RestTemplateProperties(
    @NotEmpty val maxTotalPooling: Int,
    @NotEmpty val defaultMaxPerRoute: Int,
    @NotEmpty val requestTimeout: Int,
    @NotEmpty val poolTimeout: Int,
    @NotEmpty val connectionTimeout: Int
)
